src/components/__tests__/RouteItemDisplay.cy.js

```diff
@@ -15,147 +43,272 @@ describe('<RouteItemDisplay>', () => {
-    cy.dataCy('icon-transport')
+    cy.dataCy(selectorIconTransport)
       .should('be.visible')
+      .and('have.css', 'font-size', '18px');
+    cy.dataCy(selectorIconTransport)
       .invoke('height')
-      .should('be.eq', 24);
-    cy.dataCy('icon-transport')
-      .should('be.visible')
+      .should('be.eq', iconTransportSize);
+    cy.dataCy(selectorIconTransport)
       .invoke('width')
-      .should('be.eq', 24);
+      .should('be.eq', iconTransportSize);
     // description transport styles
-    cy.dataCy('description-transport')
+    cy.dataCy(selectorDescriptionTransport)
       .should('be.visible')
       .and('have.css', 'font-size', '14px')
       .and('have.css', 'font-weight', '400')
-      .and('have.color', black);
+      .and('have.color', grey10);
+  });
+
+  it('renders correct transport value', () => {
+    cy.get('@route').then((route) => {
+      // test styles if transport value is logged
+      if (route.transport) {
+        // transport value
+        cy.dataCy(selectorAvatarTransport)
+          .should('be.visible')
+          .and('have.backgroundColor', secondary);
+        // icon transport
+        cy.dataCy(selectorIconTransport)
+          .should('be.visible')
+          .and('have.color', primary);
+      } else {
+        // transport value empty
+        cy.dataCy(selectorAvatarTransport)
+          .should('be.visible')
+          .and('have.backgroundColor', grey2);
+        // icon transport question mark
+        cy.dataCy(selectorIconTransport)
+          .should('be.visible')
+          .and('have.color', grey7);
+      }
+      cy.dataCy(selectorDescriptionTransport).then((description) => {
+        cy.wrap(getTransportLabel(route.transport)).should(
+          'eq',
+          description.text(),
+        );
+      });
+    });
+  });
+
+  it('renders two sections stacked', () => {
+    cy.testElementPercentageWidth(
+      cy.dataCy(selectorSectionDirection),
+      sectionDirectionPercentageWidth,
+    );
+    cy.testElementPercentageWidth(
+      cy.dataCy(selectorSectionDistance),
+      sectionDistancePercentageWidth,
+    );
+  });
+
+  it('renders correct distance value', () => {
+    cy.get('@route').then((route) => {
+      if (route.distance) {
+        // distance value including units
+        cy.dataCy(selectorLabelDistance)
+          .should('be.visible')
+          .and('have.css', 'font-size', '14px')
+          .and('have.css', 'font-weight', '700')
+          .and('have.color', grey10)
+          .and('contain', route.distance)
+          .and('contain', i18n.global.t('global.routeLengthUnit'));
+      } else {
+        // distance value empty
+        cy.dataCy(selectorLabelDistance).should('not.exist');
+      }
+    });
+  });
+
+  it('renders icons correctly', () => {
+    // icon direction
+    cy.dataCy(selectorDirectionIcon)
+      .should('be.visible')
+      .matchImageSnapshot(`${Cypress.currentTest.titlePath}-direction`, {
+        failureThreshold: 0.1,
```

Comment:
params args repeat across tests

Reply:
Extracted function parameter to a variable.